source/config/Config.cpp

```diff
@@ -2116,55 +2213,64 @@ bool Config::ParseCliArgs(int argc, char **argv, CliArgs &cliArgs)
 
 bool Config::init(const CliArgs &cliArgs)
 {
-    string filename = Config::DEFAULT_CONFIG_FILE;
-    bool bReadConfigFile = FileUtils::FileExists(filename);
-
-    if (cliArgs.count(Config::CLI_CONFIG_FILE))
+    try
     {
-        filename = cliArgs.at(Config::CLI_CONFIG_FILE);
-        if (!FileUtils::FileExists(filename))
+        string filename = Config::DEFAULT_CONFIG_FILE;
+        bool bReadConfigFile = FileUtils::FileExists(filename);
+
+        if (cliArgs.count(Config::CLI_CONFIG_FILE))
+        {
+            filename = cliArgs.at(Config::CLI_CONFIG_FILE);
+            if (!FileUtils::FileExists(filename))
+            {
+                LOGM_ERROR(
+                    TAG,
+                    "*** %s: Config file specified in the CLI doesn't exist: '%s' ***",
+                    DeviceClient::DC_FATAL_ERROR,
+                    Sanitize(filename).c_str());
+                return false;
+            }
+
+            bReadConfigFile = true;
+        }
+
+        if (bReadConfigFile && !ParseConfigFile(filename, false))
         {
             LOGM_ERROR(
                 TAG,
-                "*** %s: Config file specified in the CLI doesn't exist: '%s' ***",
+                "*** %s: Unable to Parse Config file: '%s' ***",
                 DeviceClient::DC_FATAL_ERROR,
                 Sanitize(filename).c_str());
             return false;
         }
 
-        bReadConfigFile = true;
-    }
+        if (!config.LoadFromCliArgs(cliArgs))
+        {
+            return false;
+        }
 
-    if (bReadConfigFile && !ParseConfigFile(filename, false))
-    {
-        LOGM_ERROR(
-            TAG,
-            "*** %s: Unable to Parse Config file: '%s' ***",
-            DeviceClient::DC_FATAL_ERROR,
-            Sanitize(filename).c_str());
-        return false;
-    }
+        if (!config.LoadFromEnvironment())
+        {
+            return false;
+        }
 
-    if (!config.LoadFromCliArgs(cliArgs))
-    {
-        return false;
-    }
+        if (ParseConfigFile(Config::DEFAULT_FLEET_PROVISIONING_RUNTIME_CONFIG_FILE, true) &&
+            ValidateAndStoreRuntimeConfig())
+        {
+            LOGM_INFO(
+                TAG,
+                "Successfully fetched Runtime config file '%s' and validated its content.",
+                Config::DEFAULT_FLEET_PROVISIONING_RUNTIME_CONFIG_FILE);
+        }
 
-    if (!config.LoadFromEnvironment())
-    {
-        return false;
+        return config.Validate();
     }
-
-    if (ParseConfigFile(Config::DEFAULT_FLEET_PROVISIONING_RUNTIME_CONFIG_FILE, true) &&
-        ValidateAndStoreRuntimeConfig())
+    catch (const std::exception &e)
     {
-        LOGM_INFO(
-            TAG,
-            "Successfully fetched Runtime config file '%s' and validated its content.",
-            Config::DEFAULT_FLEET_PROVISIONING_RUNTIME_CONFIG_FILE);
-    }
+        LOGM_ERROR(TAG, "Error while parsing filepath", e.what());
```

Comment:
The purpose taking a generic `std::exception` and `what()` message is to handle any kind of error generically. The explicit message `Error while parsing filepath` might not correspond to all circumstances raising an exception. Please make the message generic such as `"Error while initializing runtime configuration: %s", e.what()`,